src/qt/bitcoingui.cpp

```diff
@@ -1084,6 +1091,20 @@ void BitcoinGUI::setEncryptionStatus(int status)
 }
 #endif // ENABLE_WALLET
 
+void BitcoinGUI::updateProxyIcon()
+{
+    bool proxy_enabled = false;
+    proxyType proxy;
+
+    for (int n = 0; !proxy_enabled && n < NET_MAX; ++n) {
+        proxy_enabled = GetProxy((Network) n, proxy);
```

Comment:
Instead of directly accessing core features, can you wrap that via clientmodel?
There are serval tries to detach the GUI from the rest and those little things will cause a lot of work.